contrib/platform/src/com/sun/jna/platform/win32/Wevtapi.java

```diff
@@ -0,0 +1,702 @@
+     * @param Timeout    [in] Reserved. Must be zero.
+     * @param Flags      [in]Reserved. Must be zero.
+     * @return If successful, the function returns a session handle that you can use to access event log information
+     * on the remote computer; otherwise, NULL. If NULL, call {@link Kernel32#GetLastError} function to get
+     * the error code.
+     */
+    EVT_HANDLE EvtOpenSession(int LoginClass, Winevt.EVT_RPC_LOGIN Login, int Timeout, int Flags);
+
+    /**
+     * Closes an open handle.
+     * https://msdn.microsoft.com/en-us/library/windows/desktop/aa385344(v=vs.85).aspx
+     *
+     * @param Object [in] An open event handle to close.
+     * @return True The function succeeded, False The function failed. To get the error code,
+     * call the {@link Kernel32#GetLastError} function.
+     */
+    boolean EvtClose(EVT_HANDLE Object);
+
+    /**
+     * Cancels all pending operations on a handle.
+     * https://msdn.microsoft.com/en-us/library/windows/desktop/aa385335(v=vs.85).aspx
+     *
+     * @param Object The handle whose operation you want to cancel. You can cancel the following operations:
+     *               <ul>
+     *               <li>{@link Wevtapi#EvtClearLog}</li>
+     *               <li>{@link Wevtapi#EvtExportLog}</li>
+     *               <li>{@link Wevtapi#EvtNext}</li>
+     *               <li>{@link Wevtapi#EvtQuery}</li>
+     *               <li>{@link Wevtapi#EvtSeek}</li>
+     *               <li>{@link Wevtapi#EvtSubscribe}</li>
+     *               </ul>
+     *               To cancel the {@link Wevtapi#EvtClearLog}, {@link Wevtapi#EvtExportLog}, {@link Wevtapi#EvtQuery},
+     *               and EvtSubscribe operations, you must pass the session handle. To specify the default
+     *               session (local session), set this parameter to NULL.
+     * @return True The function succeeded, False The function failed. To get the error code, call
+     * the {@link Kernel32#GetLastError} function.
+     */
+    boolean EvtCancel(EVT_HANDLE Object);
+
+    /**
+     * Gets a text message that contains the extended error information for the current error.
+     * https://msdn.microsoft.com/en-us/library/windows/desktop/aa385380(v=vs.85).aspx
+     *
+     * @param BufferSize [in] The size of the Buffer buffer, in characters.
+     * @param Buffer     [in] A caller-allocated string buffer that will receive the extended error information.
+     *                   You can set this parameter to NULL to determine the required buffer size.
+     * @param BufferUsed [out] The size, in characters, of the caller-allocated buffer that the function used or
+     *                   the required buffer size if the function fails with ERROR_INSUFFICIENT_BUFFER.
+     * @return The return value is ERROR_SUCCESS if the call succeeded; otherwise, a Win32 error code.
+     */
+    int EvtGetExtendedStatus(int BufferSize, Pointer Buffer, IntByReference BufferUsed);
```

Comment:
It should be save to use a char[] as the buffer, as this better aligns with the usage and is the default mapping. If a char[] is used, this works as expected:

```java
char[] buffer = new char[1024];
EvtGetExtendedStatus(buffer.length, buffer, null);
String result = Native.toString(buffer):
```

Expected in this case means: with bare memory you'd have to consider the width/size of wchar so multiply the value of BufferUsed by two to allocate the right amount of memory.